enterprise/kernel/src/main/java/org/neo4j/kernel/enterprise/builtinprocs/EnterpriseBuiltInDbmsProcedures.java

```diff
@@ -527,4 +524,14 @@ public QueryTerminationResult( QueryId queryId, String username )
             this.connectionCount = connectionCount;
         }
     }
+
+    public static class MetadataResult
+    {
+        public final Map<String,Object> metadata;
```

Comment:
I'm not sure of the original use case for this feature, but I would have thought the result table might look nice as a two column table, with key and value pairs, instead of a single column, single row with a single map value. Just a suggestion.